homework/index.js

```diff
@@ -29,26 +37,163 @@
+        lastUpdateTime: getDateTimeText(repositoryInfo.updated_at),
+        contributorsUrl: repositoryInfo.contributors_url,
+      }));
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const repoDetailsListItem = createAndAppend('li', ul, {
+      class: 'repository-item',
+    });
+    const repoDetailsTable = createAndAppend('table', repoDetailsListItem, {
+      class: 'repository-item-table',
+    });
+    appendRepoDetail('Repository:', repo.name, repoDetailsTable, repo.url);
+    appendRepoDetail('Description:', repo.description, repoDetailsTable);
+    appendRepoDetail('Forks:', repo.forks, repoDetailsTable);
+    appendRepoDetail('Updated:', repo.lastUpdateTime, repoDetailsTable);
+  }
+
+  function renderContributorDetails(contributor, ul, index) {
+    const repoContributorListItem = createAndAppend('li', ul, {
+      class: 'contributors-item',
+    });
+    if (index === 0) {
+      createAndAppend('h6', repoContributorListItem, {
+        text: 'Contributions:',
+      });
+    }
+    createAndAppend('img', repoContributorListItem, {
+      src: contributor.avatar_url,
+      alt: 'Contributor avatar picture',
+    });
+    createAndAppend('a', repoContributorListItem, {
+      href: contributor.html_url,
+      text: contributor.login,
+      target: '_blank',
+    });
+    createAndAppend('span', repoContributorListItem, {
+      text: contributor.contributions,
+    });
+  }
+
+  function renderHeaderSection(title, parent) {
+    const header = createAndAppend('header', parent);
+    createAndAppend('span', header, {
+      text: title,
+      class: 'header-title',
+    });
+    return createAndAppend('select', header);
+  }
+
+  function renderSubSection(sectionId, parent) {
+    const subSection = createAndAppend('section', parent, {
+      class: sectionId,
+      id: sectionId,
+    });
+    createAndAppend('ul', subSection, { class: 'sub-container' });
+  }
+
+  function renderMainSectionWithSubSections(parent) {
+    const mainSection = createAndAppend('main', parent, { class: 'container' });
+    renderSubSection(REPOSITORY_DETAIL_SECTION_ID, mainSection);
+    renderSubSection(REPOSITORY_CONTRIBUTORS_SECTION_ID, mainSection);
+  }
+
+  function getUnorderedListAsEmpty(sectionId) {
+    const section = document.getElementById(sectionId);
+    const ul = section.getElementsByTagName('ul')[0];
+    ul.innerHTML = ''; // If the ul has child nodes, clear them first
+    return ul;
+  }
+
+  function renderRepositoryDetailSection(repository) {
+    renderRepoDetails(
+      repository,
+      getUnorderedListAsEmpty(REPOSITORY_DETAIL_SECTION_ID),
+    );
+  }
+
+  function renderRepositoryContributorsSection(repository) {
+    const ul = getUnorderedListAsEmpty(REPOSITORY_CONTRIBUTORS_SECTION_ID);
+    fetchJSON(repository.contributorsUrl).then(contributors => {
+      // In case of non contributors situation
+      if (contributors) {
```

Comment:
If there are no contributors, you are not rendering anything on the right-hand side. This may leave users wondering why they don't see anything there. I think rendering just an empty list would better